nvtabular/column.py

```diff
@@ -0,0 +1,75 @@
+from dataclasses import dataclass, field
+from typing import Dict, List, Optional, Text
+
+import tree
```

Comment:
You're totally right. I mentioned this to @benfred yesterday, at this point I am not even sure whether this `Columns` class makes sense. I just created it to replace `ColumnNames` and add some utility methods to make it easy to use and easy to change the `fit` & `transform` method of each op. I wanted to cut some corners to have the option of merging a PR this week and then iterate on some parts of the design later on, this because quite a few things like the schema stuff would depend on some sort of column tagging. I would also be fine with the approach of hashing out the full design of this first, but at the very least having cut these corners allows us to make a decision which approach to take.

You could argue that we already have a `ColumnGroup` class that conceptually is kinda the same as `Columns`, so it might be better to merge the 2 classes. `ColumnGroup` is an interesting class since it also represents a node in the transform-DAG. So an option could be to break out parts of that class in a base-class and potentially put that in a merlin-utils library that can be used in various places. That would make it much easier to use libraries like Transformers4Rec without NVTabular.

Reply:
Yeah, that's a good point re: `ColumnGroup` filling multiple conceptual roles as both a collection of columns and a workflow graph node. 🤔

Reply:
I think the version of `ColumnGroup` in Transformers4Rec is roughly the core of a column-grouping class that isn't also a graph node. I'm not entirely sure how to sort out `Column` vs `Columns` vs `ColumnGroup` vs `ColumnGroupNode`, but I'll keep pondering it.